From 844d3c3179776f3ff73c527bf31041baa86af256 Mon Sep 17 00:00:00 2001
From: Sai Vemuri <svemuri@redhat.com>
Date: Thu, 11 Sep 2014 22:43:01 -0400
Subject: [infiniband] cxgb4: Drop RX_DATA packets if the endpoint is gone

Message-id: <1410475447-94392-43-git-send-email-svemuri@redhat.com>
Patchwork-id: 93825
O-Subject: [RHEL7.1 PATCH BZ 1124947 042/108] RDMA/cxgb4: Drop RX_DATA packets if the endpoint is gone
Bugzilla: 1124947
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Signed-off-by: Steve Wise <swise@opengridcomputing.com>
Signed-off-by: Roland Dreier <roland@purestorage.com>
(cherry picked from commit 977116c69862a6062f302395cb3546544d7e1bc1)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/infiniband/hw/cxgb4/cm.c b/drivers/infiniband/hw/cxgb4/cm.c
index 9e1cb27..e5e438f 100644
--- a/drivers/infiniband/hw/cxgb4/cm.c
+++ b/drivers/infiniband/hw/cxgb4/cm.c
@@ -1521,6 +1521,8 @@ static int rx_data(struct c4iw_dev *dev, struct sk_buff *skb)
  __u8 status = hdr->status;
 
  ep = lookup_tid(t, tid);
+ if (!ep)
+  return 0;
  PDBG("%s ep %p tid %u dlen %u\n", __func__, ep, ep->hwtid, dlen);
  skb_pull(skb, sizeof(*hdr));
  skb_trim(skb, dlen);
-- 
1.7.1